core/src/main/java/io/nixer/nixerplugin/core/detection/filter/login/UserAgentFailedLoginOverThresholdFilter.java

```diff
@@ -18,21 +16,26 @@
  */
 public class UserAgentFailedLoginOverThresholdFilter extends MetadataFilter {
 
-    private final UserAgentTokenizer userAgentTokenizer = UserAgentTokenizer.sha1Tokenizer();
+    private final UserAgentTokenizer userAgentTokenizer;
     private final UserAgentOverLoginThresholdRegistry userAgentOverLoginThresholdRegistry;
 
-    public UserAgentFailedLoginOverThresholdFilter(UserAgentOverLoginThresholdRegistry userAgentOverLoginThresholdRegistry) {
-        Assert.notNull(userAgentOverLoginThresholdRegistry, "UsernameOverLoginThresholdRegistry must not be null");
+    private final RequestMatcher requestMatcher = new AntPathRequestMatcher("/login", "POST");
```

Comment:
Hardcoding these values has serious implications. It can prevent the filter from working for custom login endpoint.

It should be at for example injected in _DetectionConfiguration_. Login URL endpoint probably can be extracted from Spring Security? If not, a configuration in application.properties is a better option than a hardcode values.

Reply:
I totally agree that instantiating the matcher in this place is a bad practice. 
I did it that way here temporarily as I am going to do a major cleanup of the matchers (including other filters) as a separate PR.

Reply:
In that case - do we need to define them here? I would rather remove them completely, which would impact the performance but the code will still be working for non-standard endpoints...